tokio/src/io/util/copy.rs

```diff
@@ -84,6 +84,11 @@ impl CopyBuffer {
                 }
             }
 
+            // If pos larger than cap, this loop will never stop.
+            // In particular, user's wrong poll_write implementation returning
+            // incorrect written length may lead to thread blocking.
+            debug_assert!(self.pos <= self.cap, "pos should never larger than cap");
```

Comment:
Seems like we should give the user an better error message.
```suggestion
            debug_assert!(self.pos <= self.cap, "writer returned length larger than input slice");
```

Reply:
done